mlflow/transformers/llm_inference_utils.py

```diff
@@ -12,6 +21,167 @@
+                    Property("completion_tokens", DataType.long),
+                    Property("total_tokens", DataType.long),
+                ]
+            ),
+        ),
+    ]
+)
+
+_LLM_INFERENCE_CHAT_OUTPUT_SCHEMA = Schema(
+    [
+        ColSpec(
+            name="message",
+            type=Object(
+                [
+                    Property("role", DataType.string),
+                    Property("content", DataType.string),
+                ]
+            ),
+        ),
+        ColSpec(
+            name="finish_reason",
+            type=DataType.string,
+        ),
+        ColSpec(
+            name="usage",
+            type=Object(
+                [
+                    Property("prompt_tokens", DataType.long),
+                    Property("completion_tokens", DataType.long),
+                    Property("total_tokens", DataType.long),
+                ]
+            ),
+        ),
+    ]
+)
+
+_SIGNATURE_FOR_LLM_INFERENCE_TASK = {
+    _LLM_INFERENCE_TASK_CHAT: ModelSignature(
+        inputs=CHAT_MODEL_INPUT_SCHEMA, outputs=_LLM_INFERENCE_CHAT_OUTPUT_SCHEMA
+    ),
+    _LLM_INFERENCE_TASK_COMPLETIONS: ModelSignature(
+        inputs=COMPLETIONS_MODEL_INPUT_SCHEMA, outputs=_LLM_INFERENCE_COMPLETIONS_OUTPUT_SCHEMA
+    ),
+}
+
+
+def infer_signature_from_llm_inference_task(
+    inference_task: str, signature: Optional[ModelSignature] = None
+) -> ModelSignature:
+    """
+    Infers the signature according to the MLflow inference task.
+    Raises exception if a signature is given.
+    """
+    if signature is not None:
+        raise MlflowException(
+            f"When `task` is specified as `{inference_task}`, the signature would "
+            "be set by MLflow. Please do not set the signature."
+        )
+
+    return _SIGNATURE_FOR_LLM_INFERENCE_TASK[inference_task]
+
+
+def convert_data_messages_with_chat_template(data, tokenizer):
+    """For the Chat inference task, apply chat template to messages to create prompt."""
+    try:
+        messages = data.pop("messages").tolist()[0]
+    except KeyError:
+        raise MlflowException("The 'messages' field is required for the Chat inference task.")
+
+    try:
+        messages_str = tokenizer.apply_chat_template(
+            messages, tokenize=False, add_generation_prompt=True
+        )
+    except Exception as e:
+        raise MlflowException(f"Failed to apply chat template: {e}")
+
+    data["prompt"] = messages_str
+
+
+def preprocess_llm_inference_params(
+    data,
+    flavor_config: Optional[Dict[str, Any]] = None,
+) -> Dict[str, Any]:
+    """
+    When a MLflow inference task is given, return updated `data` and `params` that
+    - Extract the parameters from the input data.
+    - Replace OpenAI specific parameters with Hugging Face specific parameters, in particular
+      - `max_tokens` with `max_new_tokens`
+      - `stop` with `stopping_criteria`
+    """
+    updated_data = pd.DataFrame()
```

Comment:
I think we can simply use dictionary, datafrmae will be converted to dict anyway

```
        if isinstance(data, pd.DataFrame):
            input_data = self._convert_pandas_to_dict(data)
```